vault/ha.go

```diff
@@ -388,87 +388,122 @@ func (c *Core) StepDown(httpCtx context.Context, req *logical.Request) (retErr e
 	return retErr
 }
 
+func (c *Core) restart() {
+	select {
+	case c.restartCh <- struct{}{}:
+	default:
+		c.logger.Warn("ignoring restart request: restart is already in progress")
+	}
+}
+
+func (c *Core) drainPendingRestarts() {
+	for {
+		select {
+		case <-c.restartCh:
+			c.logger.Warn("ignoring restart request: restart is already in progress")
+
+		default:
+			return
+		}
+	}
+}
+
 // runStandby is a long running process that manages a number of the HA
 // subsystems.
-func (c *Core) runStandby(doneCh, manualStepDownCh, stopCh chan struct{}) {
+func (c *Core) runStandby(doneCh, manualStepDownCh, stopCh chan struct{}, restartCh chan struct{}) {
```

Comment:
```suggestion
//
// doneCh is used to indicate that the standby has finished operating in this invocation
// manualStepDownCh is a channel passed to the wait for leader process, to allow interrupting leadership acquisition by this node
// stopCh can be used to stop the standby and shutdown the process
// restartCh can be used to gracefully reset standby node state, reloading all cached information
//
// stopCh and restartCh differ in that the former is terminal and the latter is re-entrant.
func (c *Core) runStandby(doneCh, manualStepDownCh, stopCh chan struct{}, restartCh chan struct{}) {
```

@phil9909 WDYT? In particular, I'm trying to make sure we have some inline documentation in useful places for what these channels do and the differences between them.

Reply:
Ok, documenting the channels purpose makes sense to me. I understood you wanted to document why we loop inside `runStandby` and not outside of it. 

I took your proposal and added some more details to it (mainly about how the channel will be used i.e. does `runStandby` close it and/or read from it) and added directions to the channels were possible (`chan` vs `<-chan` vs `chan<-`) to use the type system as documentation. See https://github.com/openbao/openbao/commit/fedeb2455c198e954111f0674e8c3b7d0dbecea4